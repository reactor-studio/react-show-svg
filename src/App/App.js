import React, { Component } from 'react';
import PropTypes from 'prop-types';
import svgtojsx from 'svg-to-jsx';

class ReactShowSvg extends Component {
  constructor(props) {
    super(props);

    this.state = {
      icon: null
    };
  }

  componentDidMount() {
    this.svgToJsx(this.props.icon);
  }

  svgToJsx = (raw) => {
    return this.parseSvg(raw).then((jsx) => {
      const icon = this.renderIcon(jsx);

      this.setState({ icon });
    });
  }

  parseSvg = (raw) => {
    const { size, width, height, fill } = this.props;
    const defaultSize = size || '20px';
    const defaultFill = 'black';

    return svgtojsx(raw, {
      props: {
        width: width || defaultSize,
        height: height || defaultSize,
        fill: fill || defaultFill
      }
    }).then(jsx => jsx);
  }

  renderIcon(jsx) {
    const {
      id,
      ref,
      className,
      onClick
    } = this.props;

    const icon = (
      <div
        id={id}
        ref={ref}
        onClick={onClick}
        className={className}
        dangerouslySetInnerHTML={{ __html: jsx }}
      />
    );

    return icon;
  }

  render() {
    const { icon } = this.state;

    return icon ? icon : null;
  }
}

ReactShowSvg.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.string,
  width: PropTypes.string,
  fill: PropTypes.string,
  height: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
  id: PropTypes.string,
  ref: PropTypes.func
};

export default ReactShowSvg;
