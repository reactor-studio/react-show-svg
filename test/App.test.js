import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Renderer from 'react-test-renderer';

// Component
import App from '../src/App/App';

// Config
configure({ adapter: new Adapter() });

// Assets
const icon = '<svg version="1.1"><text font-family="Verdana"/></svg>';
const parsedIcon = '<svg version="1.1"><text fontFamily="Verdana"/></svg>';

describe('App Component', () => {
  const wrapper = shallow(
    <App
      id="test-id"
      onClick={() => console.log('Good')}
      className="test-class"
      icon={icon}
    />
  );

  it('should parse svg', () => {
    wrapper.instance().parseSvg(icon).then(parsed => {
      expect(parsed).toEqual(parsedIcon);
    });
  });

  it('should create wrapper around svg', () => {
    const wrapped = shallow(wrapper.instance().renderIcon(parsedIcon));

    expect(wrapped).toMatchSnapshot();
  });
});
