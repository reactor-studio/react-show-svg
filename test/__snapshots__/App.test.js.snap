// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should create wrapper around svg 1`] = `
<div
  className="test-class"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg version=\\"1.1\\"><text fontFamily=\\"Verdana\\"/></svg>",
    }
  }
  id="test-id"
  onClick={[Function]}
/>
`;
